tools-local/tasks/mobile.tasks/AndroidAppBuilder/ApkBuilder.cs

```diff
@@ -244,30 +240,58 @@ public class ApkBuilder
         File.Delete(apkFile);
 
-        // 5. Generate key
+        // 5. Generate key (if needed) & sign the apk
+        SignApk(alignedApk, apksigner);
+
+        Utils.LogInfo($"\nAPK size: {(new FileInfo(alignedApk).Length / 1000_000.0):0.#} Mb.\n");
+
+        return (alignedApk, packageId);
+    }
+
+    private void SignApk(string apkPath, string apksigner)
+    {
+        string defaultKey = Path.Combine(OutputDir, "debug.keystore");
+        string signingKey = string.IsNullOrEmpty(KeyStorePath) ?
+            defaultKey : Path.Combine(KeyStorePath, "debug.keystore");
 
-        string signingKey = Path.Combine(OutputDir, "debug.keystore");
-        if (!string.IsNullOrEmpty(KeyStorePath))
-            signingKey = Path.Combine(KeyStorePath, "debug.keystore");
         if (!File.Exists(signingKey))
         {
-            Utils.RunProcess(keytool, "-genkey -v -keystore debug.keystore -storepass android -alias " +
+            Utils.RunProcess("keytool", "-genkey -v -keystore debug.keystore -storepass android -alias " +
                 "androiddebugkey -keypass android -keyalg RSA -keysize 2048 -noprompt " +
                 "-dname \"CN=Android Debug,O=Android,C=US\"", workingDir: OutputDir, silent: true);
         }
-        else
+        else if (Path.GetFullPath(signingKey) != Path.GetFullPath(defaultKey))
         {
             File.Copy(signingKey, Path.Combine(OutputDir, "debug.keystore"));
         }
+        Utils.RunProcess(apksigner, $"sign --min-sdk-version {MinApiLevel} --ks debug.keystore " +
+            $"--ks-pass pass:android --key-pass pass:android {apkPath}", workingDir: OutputDir);
+    }
+
+    public void ReplaceFileInApk(string file)
+    {
+        if (string.IsNullOrEmpty(AndroidSdk))
+            AndroidSdk = Environment.GetEnvironmentVariable("ANDROID_SDK_ROOT");
 
-        // 6. Sign APK
+        if (string.IsNullOrEmpty(AndroidSdk) || !Directory.Exists(AndroidSdk))
+            throw new ArgumentException($"Android SDK='{AndroidSdk}' was not found or incorrect (can be set via ANDROID_SDK_ROOT envvar).");
 
-        Utils.RunProcess(apksigner, $"sign --min-sdk-version {MinApiLevel} --ks debug.keystore " +
-            $"--ks-pass pass:android --key-pass pass:android {alignedApk}", workingDir: OutputDir);
+        if (string.IsNullOrEmpty(BuildToolsVersion))
+            BuildToolsVersion = GetLatestBuildTools(AndroidSdk);
 
-        Utils.LogInfo($"\nAPK size: {(new FileInfo(alignedApk).Length / 1000_000.0):0.#} Mb.\n");
+        if (string.IsNullOrEmpty(MinApiLevel))
+            MinApiLevel = DefaultMinApiLevel;
 
-        return (alignedApk, packageId);
+        string buildToolsFolder = Path.Combine(AndroidSdk, "build-tools", BuildToolsVersion);
+        string aapt = Path.Combine(buildToolsFolder, "aapt");
+        string apksigner = Path.Combine(buildToolsFolder, "apksigner");
+
+        string apkPath = Directory.GetFiles(Path.Combine(OutputDir, "bin"), "*.apk").First();
```

Comment:
There would be multiple `.apk` files when building the AndroidAppBuilder with different configurations, framework, would the use case include some sort of `git clean -xdff`. Is there a straightforward way of determining which apk to target within the ApkBuilder context?